cmd/snap/cmd_validate.go

```diff
@@ -0,0 +1,191 @@
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdValidate struct {
+	clientMixin
+	Monitor    bool `long:"monitor"`
+	Enforce    bool `long:"enforce"`
+	Forget     bool `long:"forget"`
+	Positional struct {
+		ValidationSet string `positional-arg-name:"<validation-set>"`
+	} `positional-args:"yes"`
+	colorMixin
+}
+
+var shortValidateHelp = i18n.G("List or set snap validations")
+var longValidateHelp = i18n.G(`
+The validate command lists validation sets or sets validations")
+`)
+
+func init() {
+	addCommand("validate", shortValidateHelp, longValidateHelp, func() flags.Commander { return &cmdValidate{} }, colorDescs.also(map[string]string{
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"monitor": i18n.G("Monitor the given validations set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"enforce": i18n.G("Enforce the given validation set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"forget": i18n.G("Forget the given validation set"),
+	}), []argDesc{{
+		// TRANSLATORS: This needs to begin with < and end with >
+		name: i18n.G("<validation-set>"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		desc: i18n.G("Validation set with an optional pinned sequence point, i.e. account/name[=seq]"),
+	}})
+}
+
+// this is reused for both account and set name of "account/name" argument
+var validName = regexp.MustCompile("^[0-9a-z]+$")
+
+func splitValidationSetArg(arg string) (account, name string, seq int, err error) {
+	parts := strings.Split(arg, "=")
+	if len(parts) > 2 {
+		return "", "", 0, fmt.Errorf("cannot parse validation set, expected account/name=seq")
+	}
+	if len(parts) == 2 {
+		seq, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return "", "", 0, err
+		}
+	}
+
+	parts = strings.Split(parts[0], "/")
+	if len(parts) != 2 {
+		return "", "", 0, fmt.Errorf("expected a single account/name")
+	}
+
+	account = parts[0]
+	name = parts[1]
+	if account == "" {
+		return "", "", 0, fmt.Errorf("invalid empty account name")
+	}
+	if !validName.MatchString(account) {
+		return "", "", 0, fmt.Errorf("invalid account name %q", account)
+	}
+
+	if name == "" {
+		return "", "", 0, fmt.Errorf("invalid empty name")
+	}
+	if !validName.MatchString(name) {
+		return "", "", 0, fmt.Errorf("invalid name %q", name)
```

Comment:
I would use 
```suggestion
		return "", "", 0, fmt.Errorf("invalid validation set name %q", name)
```

to differentiate that from the account name.

Reply:
Yeah, I think this makes sense.

Reply:
I think this may be confusing; perhaps the earlier message should be changed to "invalid account id %q". The issue is we use "account" (account id) and "name" in other places, and validation set name is refers to the combined "account-id/name".

Reply:
do you have  a place where we use "validation set name" in that sense?  that's would be validation set key really, but I' haven't found a place we use that terminology

Reply:
Sounds fine, name is probably okay, let's not overcomplicate it :)

Reply:
> do you have a place where we use "validation set name" in that sense? that's would be validation set key really, but I' haven't found a place we use that terminology

You're right, I can't find such place either, and yes, I think we use validation set key (at least internally). I'm ok to change the message as originally suggested by zyga, sounds ok?

Reply:
yes, sounds ok

Reply:
Updated.